src/helpers.js

```diff
@@ -176,14 +176,7 @@ function cloner(deep, item) {
   // Array
   if (type.array(item)) {
     if (deep) {
-      const a = [];
```

Comment:
Have you trie to replace this line by:

```js
const a = new Array(item.length);
```

?

Reply:
Because if the fact that the array does not have a pre-determined length before iterating, this should actually be even faster than `.map`.

I know v8 worked a lot to optimize callback iterations such as `.forEach` `.map` etc. but in my experience, for loops still have a significant edge.

Reply:
Also, recursive function calls have a cost and if we want to really optimize this aggressively, we can also drop recursivity by using a stack (but let's handle your modification before).

Reply:
You right, is better.

IE11
![image](https://user-images.githubusercontent.com/4757846/32593442-fd5a6be6-c538-11e7-98b0-8b7901ecffb1.png)

Chrome 63
![image](https://user-images.githubusercontent.com/4757846/32593463-1465feae-c539-11e7-8dc8-bcd0e8c55f63.png)